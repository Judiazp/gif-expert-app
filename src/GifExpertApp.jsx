import React, { useEffect, useState } from 'react';
import AddCategory from './components/addCategory';
import GifGrid from './components/gifGrid';

const GiffExpertApp = ({ defaultCategories=['']}) => {

    const initialStateDarkMode =  JSON.parse(localStorage.getItem('darkMode')) || false
    const [darkMode, setDarkMode] = useState(initialStateDarkMode)

    const [categories, setCategories] = useState(defaultCategories);
    
    useEffect(() => {
        localStorage.setItem('darkMode', JSON.stringify(darkMode))
    }, [darkMode])

    if (darkMode) {
        document.body.classList.add('dark')
    } else {
        document.body.classList.remove('dark') 
    }

    const handleClick = () => {
        setDarkMode(!darkMode)
        document.body.classList.toggle('dark')
    }

    return (
        <div id="addCategory" className="contenedor">
            <nav>
                <a class="logotipo" href="/">GiffExpertApp</a>
                <div>
                    <AddCategory setCategories={ setCategories }/>
                </div>
                <button className={ darkMode? "switch active" : "switch" } onClick={ () => handleClick()}> 
                    <span> <i class="fas fa-sun"></i> </span>
                    <span> <i class="fas fa-moon"></i> </span>
                </button>
            </nav>
            {
                categories.map(category => <GifGrid key={category} category={ category }/>)
            }
        </div>
    )
}

export default GiffExpertApp;