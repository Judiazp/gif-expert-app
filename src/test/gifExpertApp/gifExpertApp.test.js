import { shallow } from 'enzyme';
import React from 'react'
import GifExpertApp from '../../GifExpertApp'

describe('Pruebas en <GifExpertApp />', () => {
   
    test('Debe mostrar <GifExpertApp />', () => {
        const wrapper = shallow( <GifExpertApp />)
        expect( wrapper ).toMatchSnapshot()
    });

    test('Debe mostrar una lista de categorias', () => {
        const categories =['spiderman', 'hello']
        const wrapper = shallow( <GifExpertApp defaultCategories={ categories } />)
        expect( wrapper ).toMatchSnapshot()
        expect( wrapper.find('GifGrid').length ).toBe( categories.length )
    });
})