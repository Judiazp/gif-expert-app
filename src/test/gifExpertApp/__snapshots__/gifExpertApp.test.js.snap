// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifExpertApp /> Debe mostrar <GifExpertApp /> 1`] = `
<div
  className="contenedor"
  id="addCategory"
>
  <nav>
    <a
      class="logotipo"
      href="/"
    >
      GiffExpertApp
    </a>
    <div>
      <AddCategory
        setCategories={[Function]}
      />
    </div>
    <button
      className="switch"
      onClick={[Function]}
    >
      <span>
         
        <i
          class="fas fa-sun"
        />
         
      </span>
      <span>
         
        <i
          class="fas fa-moon"
        />
         
      </span>
    </button>
  </nav>
  <GifGrid
    category=""
    key=""
  />
</div>
`;

exports[`Pruebas en <GifExpertApp /> Debe mostrar una lista de categorias 1`] = `
<div
  className="contenedor"
  id="addCategory"
>
  <nav>
    <a
      class="logotipo"
      href="/"
    >
      GiffExpertApp
    </a>
    <div>
      <AddCategory
        setCategories={[Function]}
      />
    </div>
    <button
      className="switch"
      onClick={[Function]}
    >
      <span>
         
        <i
          class="fas fa-sun"
        />
         
      </span>
      <span>
         
        <i
          class="fas fa-moon"
        />
         
      </span>
    </button>
  </nav>
  <GifGrid
    category="spiderman"
    key="spiderman"
  />
  <GifGrid
    category="hello"
    key="hello"
  />
</div>
`;
